src/core/meshcop/border_agent.cpp

```diff
@@ -601,7 +631,7 @@ void BorderAgent::HandleConnected(bool aConnected)
     if (aConnected)
     {
         LogInfo("Commissioner connected");
-        mState = kStateActive;
+        mState = kStateConnectedPetition;
         mTimer.Start(kKeepAliveTimeout);
```

Comment:
Here is when secure session is established, no requirement of `kKeepAliveTimeout`. 
May move  mTimer.Start() from here to where petition/keepalive is expected, probably into `ForwardToLeader`?

Reply:
This may be a worthwhile enhancement to consider, but it's not directly related to this PR's intended goal. It's generally better to include such changes in their own dedicated PRs.

I understand the desire to enhance or update other things while writing or reviewing code, but my general advice to you and the team is to always keep PRs focused and small. See ["In praise of small pull requests"](https://testing.googleblog.com/2024/07/in-praise-of-small-pull-requests.html) (`go/tott/713` google internal link). 😃. Please feel free to submit a separate PR with your proposed changes.